src/AudioTrackManager.test.ts

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable dot-notation */
+import MockMediaAudio from "./__mocks__/MockMediaAudio";
+
+import AudioTrackManager from "./AudioTrackManager";
+import { AudioTrackBehaviour } from "./types";
+
+jest.mock("./MediaAudio", () => ({
+  __esModule: true,
+  default: jest.fn().mockImplementation(() => MockMediaAudio),
+}));
+
+describe("AudioTrackManager", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("should initialize with isPlaying as false and no currentAudio", () => {
+    const audioTrackManager = new AudioTrackManager();
+
+    expect(audioTrackManager.isPlaying).toBe(false);
+    expect(audioTrackManager["currentAudio"]).toEqual([]);
+  });
+
+  test("should play new audio tracks", () => {
+    const audioTrackManager = new AudioTrackManager();
+
+    const audioTracks = [
+      {
+        url: "track1.mp3",
+        loop: false,
+        volume: 0.5,
+        behaviour: AudioTrackBehaviour.Restart,
+        stopPlaying: false,
+      },
+      {
+        url: "track2.mp3",
+        loop: true,
+        volume: 0.8,
+        behaviour: AudioTrackBehaviour.Continue,
+        stopPlaying: false,
+      },
+    ];
+    audioTrackManager.play(audioTracks);
+
+    expect(audioTrackManager.isPlaying).toBe(true);
+    expect(audioTrackManager["currentAudio"]).toHaveLength(2);
+  });
+
+  test("should not play audio if audioTracks array is empty", () => {
+    const audioTrackManager = new AudioTrackManager();
+
+    audioTrackManager.play([]);
+
+    expect(audioTrackManager.isPlaying).toBe(false);
+  });
+
+  test("should stop all currently playing audio tracks", () => {
+    const audioTrackManager = new AudioTrackManager();
+    const audioTracks = [
+      {
+        url: "track1.mp3",
+        loop: false,
+        volume: 0.5,
+        behaviour: AudioTrackBehaviour.Restart,
+        stopPlaying: false,
+      },
+      {
+        url: "track2.mp3",
+        loop: true,
+        volume: 0.8,
+        behaviour: AudioTrackBehaviour.Continue,
+        stopPlaying: false,
+      },
+    ];
+    audioTrackManager.play(audioTracks);
+    audioTrackManager.stopAll();
+
+    expect(audioTrackManager.isPlaying).toBe(false);
+    expect(audioTrackManager["currentAudio"]).toEqual([]);
+    expect(MockMediaAudio.pause).toHaveBeenCalledTimes(2);
+  });
+
+  test("should toggle mute on all currently playing audio tracks", () => {
+    const audioTrackManager = new AudioTrackManager();
+    const audioTracks = [
+      {
+        url: "track1.mp3",
+        loop: false,
+        volume: 0.5,
+        behaviour: AudioTrackBehaviour.Restart,
+        stopPlaying: false,
+      },
+    ];
+    audioTrackManager.play(audioTracks);
+
+    audioTrackManager.toggleMute();
+    expect(audioTrackManager["currentAudio"][0].muted).toBe(true);
+
+    audioTrackManager.toggleMute();
+    expect(audioTrackManager["currentAudio"][0].muted).toBe(false);
+  });
+
+  test("should set the volume for all currently playing audio tracks", () => {
```

Comment:
I think the volume here is problematic because it overwrites the value defined in charisma.
For example, from charisma audio tracks we could get 
strimmer.mp3 at volume 0.05 and 
country-background at volume 0.80
Now as an implementor of the sdk I set a volume control to a default, maybe 0.70.
Now both files are playing at 0.70.
Instead I think the volume should be 0.05*0.70 and 0.80*0.70